src/pages/api/opt-out-feedback.ts

```diff
@@ -0,0 +1,118 @@
+import { NextApiRequest, NextApiResponse } from 'next'
+import {
+  GoogleSpreadsheet,
+  GoogleSpreadsheetRow,
+  GoogleSpreadsheetWorksheet,
+} from 'google-spreadsheet'
+
+const FEEDBACK_SHEET_ID = process.env.FEEDBACK_SHEET_ID
+const FEEDBACK_SERVICE_EMAIL = process.env.FEEDBACK_SERVICE_EMAIL
+const FEEDBACK_PRIVATE_KEY = process.env.FEEDBACK_PRIVATE_KEY
+
+export default async function handler(
```

Comment:
Maybe not a huge deal as it could be sniffed pretty easily, but do we want to require an API key via a header here too?

Reply:
Good question. 

 Maybe a better protection is actually wiring up CORS / header sniffing so the request can only be sent from developer.hashicorp.com 🤔 ?

We did something like this on Learn's feedback form (which I drew inspiration from), and that seemingly was enough to prevent abuse: https://github.com/hashicorp/learn/blob/master/pages/api/feedback.ts#L62-L69

Reply:
Yeah, good call, CORS would be appropriate here. It wouldn't necessarily stop someone from making requests from a non-browser env if we only used the proper CORS header (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin), but your origin check is effective too.

I think any approach here would be adequate!

Reply:
I think I'll do CORS (cause we will need it to allow the requests) + the origin check (to add a barrier to abuse).